### PR TITLE
Add files:scanAppData command

### PR DESCRIPTION
**Pull request by rullzer**
Fixes https://github.com/nextcloud/server/issues/3041

Add a dedicated command to scan the special appdata dir

### CODE REVIEW

File: apps/files/lib/Command/ScanAppData.php [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+		$this->root = $rootFolder;
+		$this->config = $config;
+	}
+
+	protected function configure() {
+		parent::configure();
+
+		$this
+			->setName('files:scanAppData')
```
Comment (MorrisJobke) [inline apps/files/lib/Command/ScanAppData.php]: I would vote for `files:scan-app-data` to have a common naming schema

File: apps/files/lib/Command/ScanAppData.php [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+		# $quiet   means no print at all
+		# $verbose means full printout including statistics
+		# -q	-v	full	stat
+		#  0	 0	no	yes
+		#  0	 1	yes	yes
+		#  1	--	no	no  (quiet overrules verbose)
+		$verbose = $input->getOption('verbose');
+		$quiet = $input->getOption('quiet');
+		# restrict the verbosity level to VERBOSITY_VERBOSE
+		if ($output->getVerbosity()>OutputInterface::VERBOSITY_VERBOSE) {
```
Comment (MorrisJobke) [inline apps/files/lib/Command/ScanAppData.php]: Spaces around operator

**Comment by MorrisJobke** [comment]
And of course the autoloader is not updated. Let me address all the comments.

**Comment by MorrisJobke** [comment]
> And of course the autoloader is not updated. Let me address all the comments.

Ah no - it's in files and not needed therefore.

**Review by MorrisJobke — APPROVED** [review]
Works

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**